### PR TITLE
Cleanup v1 registry code that is not used for searching

### PR DESCRIPTION
**Pull request by thaJeztah**
See individual commits for details. This will likely conflict with https://github.com/moby/moby/pull/39371, which also removes the "fallback" errors, but I think that's causing that PR to fail (i.e. when using "insecure registries", and connecting to `https://` fails, it should continue with the next endpoint).

### CODE REVIEW

File: registry/session.go [hunk 1]
```diff
@@ -1,43 +1,26 @@
 package registry // import "github.com/docker/docker/registry"
 
 import (
-	"bytes"
-	"crypto/sha256"
-
 	// this is required for some certificates
 	_ "crypto/sha512"
```
Comment (thaJeztah) [inline registry/session.go:5]: Not sure if this is still needed; it was added in https://github.com/moby/moby/pull/5600 to address https://github.com/moby/moby/issues/5173

**Comment by thaJeztah** [comment]
Whoop! this is green (with the exception of s390x, which is tracked through https://github.com/moby/moby/issues/41561). I'll kick CI once, to see if it goes green

**Review by tiborvass — APPROVED** [review]

**Review by tonistiigi — APPROVED** [review]
SGTM

**Comment by thaJeztah** [comment]
checked with @cpuguy83 if this conflicts with some of the registry work he was working on, but should not conflict, so this one should be ready to go

**Pull request merged**